### PR TITLE
chore(asm): raw body django tainting for IAST

### PR DESCRIPTION
**Pull request by christophe-papazian**
- add request body tainting for Django framework
- add 2 tests with and without iast

It works by tainting an internal private field of Django HttpRequest `_body`

## Checklist

- [x] Change(s) are motivated and described in the PR description.
- [x] Testing strategy is described if automated tests are not included in the PR.
- [x] Risk is outlined (performance impact, potential for breakage, maintainability, etc).
- [x] Change is maintainable (easy to change, telemetry, documentation).
- [x] [Library release note guidelines](https://ddtrace.readthedocs.io/en/stable/releasenotes.html) are followed. If no release note is required, add label `changelog/no-changelog`.
- [x] Documentation is included (in-code, generated user docs, [public corp docs](https://github.com/DataDog/documentation/)).
- [x] Backport labels are set (if [applicable](https://ddtrace.readthedocs.io/en/latest/contributing.html#backporting))

## Reviewer Checklist

- [x] Title is accurate.
- [x] No unnecessary changes are introduced.
- [x] Description motivates each change.
- [x] Avoids breaking [API](https://ddtrace.readthedocs.io/en/stable/versioning.html#interfaces) changes unless absolutely necessary.
- [x] Testing strategy adequately addresses listed risk(s).
- [x] Change is maintainable (easy to change, telemetry, documentation).
- [x] Release note makes sense to a user of the library.
- [x] Reviewer has explicitly acknowledged and discussed the performance implications of this PR as reported in the benchmarks PR comment.
- [x] Backport labels are set in a manner that is consistent with the [release branch maintenance policy](https://ddtrace.readthedocs.io/en/latest/contributing.html#backporting)

### CODE REVIEW

File: tests/contrib/django/django_app/appsec_urls.py [hunk 1]
```diff
@@ -190,6 +191,21 @@ def sqli_http_request_cookie_value(request):
...
 
 
+def sqli_http_request_body(request):
+    key = "master_key"
+    if key in request.POST:
+        value = request.POST[key]
+    else:
+        value = decode_aspect(request.body)
+
+    print("VALUE", value)
```
Comment (gnufede) [inline tests/contrib/django/django_app/appsec_urls.py:201]: Print statement left behind?

File: ddtrace/appsec/handlers.py [hunk 2]
```diff
@@ -169,32 +169,43 @@ def _on_django_func_wrapped(fn_args, fn_kwargs, first_arg_expected_type):
...
-            fn_args[0].path, source_name="path", source_value=fn_args[0].path, source_origin=OriginType.PATH
+        http_req = fn_args[0]
+
+        if not isinstance(http_req.COOKIES, LazyTaintDict):
+            http_req.COOKIES = LazyTaintDict(http_req.COOKIES, origins=(OriginType.COOKIE_NAME, OriginType.COOKIE))
+        if not isinstance(http_req.GET, LazyTaintDict):
+            http_req.GET = LazyTaintDict(http_req.GET, origins=(OriginType.PARAMETER_NAME, OriginType.PARAMETER))
+        if not isinstance(http_req.POST, LazyTaintDict):
+            http_req.POST = LazyTaintDict(http_req.POST, origins=(OriginType.BODY, OriginType.BODY))
+        if not is_pyobject_tainted(getattr(http_req, "_body", None)):
```
Comment (gnufede) [inline ddtrace/appsec/handlers.py:184]: Is this if independent from the previous one? or maybe we can use an `elif` here?

**Review by juanjux — COMMENTED** [review]
LGTM once you remove the print @gnufede pointed.

File: ddtrace/appsec/handlers.py (hunk 2 above)
Comment (christophe-papazian) [reply ddtrace/appsec/handlers.py:184]: they are all independent and taint different kinds of information.

File: tests/contrib/django/django_app/appsec_urls.py (hunk 1 above)
Comment (christophe-papazian) [reply tests/contrib/django/django_app/appsec_urls.py]: fixed, thank you !

**Review by gnufede — APPROVED** [review]

**Pull request merged**